### ISSUE TITLE
Rewrite pipeline launch page

### ISSUE THREAD

**Issue by ewels** [nf-core/website#2015]
The pipeline launch page is one of the few that didn't make it into the new Astro rebuild. At the time of writing, that's still running at [oldsite.nf-co.re](http://oldsite.nf-co.re/launch) with source code lurking on the `old-site` branch [here](https://github.com/nf-core/website/blob/old-site/public_html/launch.php). We do some redirection magic to make that new website render the URL from the oldsite without anyone knowing 🤫 

We need to rewrite the nf-core web launch pages. I've previously discussed this with @mashehu but I forget the details now, so intention is to come back later and edit this issue to include more detail.

**Comment by multimeric** [comment]
Can you explain the technologies that might be used to rewrite the launch page? Because as I understand it can't all be done client side.

I might be able to help with this, assuming that the launch service can be modularised to run separately from the rest of the site, since I'm interested in hosting my own copy of the launch service only. For example if it were hosted on AWS I'd be happy to work on a lambda function that handles the backend logic, and a `Svelte` component for the frontend.

**Comment by mashehu** [comment]
nothing written in stone yet. but we would like to launch a a local server from the CLI (so people can use it in offline environments and also we don't send sensitive data around) and then reuse our tech-stack (astro+svelte) for the launcher pages (my current gut feeling is to use something like https://vite-pwa-org.netlify.app/ to do that, but haven't looked to much into it.) you are very welcome to contribute to it (I will hopefully start this endeavor this or next week.

**Comment by multimeric** [comment]
Okay that sounds great. If the web app can run locally that resolves my issue and makes it even more useful for me to contribute the feature. Can you explain why you want to re-use the tech stack if the CLI is a separate repo to the website? Is it to maintain a consistent style? 

Please let me know when the tech decisions have progressed enough for me to make a start on this, because I happen to have some available time for this now. Otherwise I can also wait for your implementation later on.

**Comment by mashehu** [comment]
yep, style consistency is one of the main reason for sticking with the current stack. Additionally, we will can hopefully reuse components from/for other pages (parameters page, schema builder page)

**Comment by ewels** [comment]
NB: The schema builder page will also run from the CLI, with a local server. So that and the rewritten launch will both be in nf-core/tools I guess.

**Comment by multimeric** [comment]
Good point. Should I close this issue and open another one in that repo?

**Comment by mashehu** [comment]
no need, thanks, we are already trying to gather everything there: https://github.com/nf-core/tools/issues/2429

**Comment by multimeric** [comment]
That task seems to be about supporting the nextflow schema, rather than the launch page? Is https://github.com/nf-core/tools/issues/2463 maybe more relevant? I would love to progress this if possible.
